File: match-service/src/main/java/fact/it/matchservice/dto/MatchRequest.java
package fact.it.matchservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class MatchRequest {
    private String homeTeamCode; // Can be team ID or name
    private String awayTeamCode; // Can be team ID or name
    private LocalDateTime dateTime;
    private String location;
    private Integer homeTeamScore;
    private Integer awayTeamScore;
}

File: soccer-frontend/src/dto/MatchRequest.ts
export interface MatchRequest {
    homeTeamCode: string;     // Can be team ID or name
    awayTeamCode: string;     // Can be team ID or name
    dateTime: string;           // TypeScript uses Date object for datetime
    location: string;
    homeTeamScore: number;   // Made optional with '?' as scores might not be known initially
    awayTeamScore: number;   // Made optional with '?' as scores might not be known initially
}